apisix/plugins/elasticsearch-logger.lua

```diff
@@ -23,18 +23,35 @@ local plugin          = require("apisix.plugin")
 
 local ngx             = ngx
 local str_format      = core.string.format
+local math_random     = math.random
+local type            = type
 
 local plugin_name = "elasticsearch-logger"
 local batch_processor_manager = bp_manager_mod.new(plugin_name)
 
 
-local schema = {
-    type = "object",
-    properties = {
-        endpoint_addr = {
+local endpoint_schema = {
+    anyOf = {
+        {
+            -- deprecated, use "array" instead
             type = "string",
             pattern = "[^/]$",
         },
+        {
+            type = "array",
+            minItems = 1,
+            items = {
+                type = "string",
+                pattern = "[^/]$",
+            },
+        }
+    }
+}
+
+local schema = {
+    type = "object",
+    properties = {
+        endpoint_addr = endpoint_schema,
```

Comment:
Better to use `endpoint_addrs` with the new type.
Using different types in the same field brings us lots of trouble when we need to work with static-type language, like writing a Go client.

Reply:
Hi @spacewander ,  thank you for your reply. 
In my opinion, to use **endpoint_addrs** will cause trouble, this change is not backward compatible.

Reply:
Keep endpoint_addr, add a new field endpoint_addrs to support both string and array better

Reply:
> this change is not backward compatible

I am confused. People can still use the old endpoint_addr configuration.

> Keep endpoint_addr, add a new field endpoint_addrs to support both string and array better

It is not a good idea to have a field of multiple types. The upstream's nodes field is an example. People keep complaining that this field doesn't have a consistent type.

Reply:
Understood. So we need to keep both endpoint_addr and endpoint_addrs, where endpoint_addrs is only specified as array type, is that right?

Reply:
Yes